packages/api-client/src/views/Request/RequestSection/RequestBody.vue

```diff
@@ -367,12 +367,22 @@ const removeBinaryFile = () =>
 function handleRemoveFileFormData(rowIdx: number) {
   const currentParams = formParams.value
   const updatedParams = [...currentParams]
-
-  if (currentParams.length > 1) {
-    // Removes the row if not the last one
+  const param = currentParams[rowIdx]
+  const file = param?.file as File | undefined
+
+  // Empty key value or non updated file name then remove the row
+  if (
+    currentParams.length > 1 &&
+    ((!param?.key && !param?.value) ||
+      (file && param?.key === file.name && param?.value === file.name))
+  ) {
```

Comment:
I would move some of these to a variable just to make it more readable, that or add comments to each line